server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -365,7 +364,7 @@ public ClusterState execute(ClusterState currentState) {
                 newEntry = SnapshotsInProgress.startedEntry(
                         new Snapshot(repositoryName, snapshotId), request.includeGlobalState(), request.partial(),
                         indexIds, dataStreams, threadPool.absoluteTimeInMillis(), repositoryData.getGenId(), shards,
-                        userMeta, version, featureStates);
+                        repository.adaptUserMetadata(request.userMetadata()), version, featureStates);
```

Comment:
Had to move the computation here so it happens after loading `RepositoryData` and we are guaranteed to have the hash computed already. Probably kind of nice to move it here anyway so the repo could theoretically run this operation based off of a recent cluster state.

Reply:
> can think of a better place we can hook in the license check (because a license check is not really suitable in the adaptUserMetadata method).

I couldn't really find any other hook during snapshotting where I could be sure that `getRepositoryData` was called for sure and the repo properly initialized. `adaptUserMetadata` may be called without any prior interaction with the repo in a number of scenarios.